`publish doc` copies the wrong directory

There are 3 subdirectories under `doc/` in the build artifacts: `_html`, `_odoc`, `_mld`. dune-release currently copies all three, instead of copying just the content of `_html`.

Additionally, it would be nice to ask confirmation where the doc should go before actually copying it and pushing.

I pinned `dune-release`.
```
$ odoc --version
1.2.0
$ dune-release publish doc --dry-run
[...]
$ ls _build/lpd-1.2.2-1-g8469692/_build/default/CHANGES.md  lpd.cmxa     lpd.ml     lpd.socket.dune   socket.a      socket.ml
LICENSE.md  lpd.cmxs*    lpd.ml.d   lpd.version.sexp  socket.cma    socket.ml.d
lpd.a       lpd.dune     lpd.mli    META.lpd          socket.cmxa   socket.mli
lpd.cma     lpd.install  lpd.mli.d  README.md         socket.cmxs*  socket.mli.d
```
so no `_doc` directory.  However, after `jbuilder build @doc`, I get
```
$ ls _build/default/_doc/
_html/  _mlds/  _odoc/
```
and  `dune-release publish` pushes the [same directory structure](https://github.com/Chris00/lpd/tree/gh-pages).

Note that, `dune-release publish` finishes with
```
dune-release: [ERROR] Could not find release id in response:
                      HTTP/1.1 404 Not Found
                      Server: GitHub.com
                      Date: Sat, 09 Jun 2018 18:40:34 GMT
                      Content-Type: application/json;
                      charset=utf-8
                      Content-Length: 120
                      Status: 404 Not Found
                      X-RateLimit-Limit: 5000
                      X-RateLimit-Remaining: 4945
                      X-RateLimit-Reset: 1528571402
                      X-OAuth-Scopes: repo
                      X-Accepted-OAuth-Scopes: repo
                      X-GitHub-Media-Type: github.v3;
                      format=json
                      Access-Control-Expose-Headers: ETag, Link, Retry-After,
                      X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining,
                      X-RateLimit-Reset, X-OAuth-Scopes,
                      X-Accepted-OAuth-Scopes,
                      X-Poll-Interval
                      Access-Control-Allow-Origin:
                      *
                      Strict-Transport-Security: max-age=31536000;
                      includeSubdomains; preload
                      X-Frame-Options: deny
                      X-Content-Type-Options: nosniff
                      X-XSS-Protection: 1; mode=block
                      Referrer-Policy: origin-when-cross-origin,
                      strict-origin-when-cross-origin
                      Content-Security-Policy: default-src
                      'none'
                      X-Runtime-rack: 0.028666
                      X-GitHub-Request-Id:
                      A740:57C5:3BDE7BF:7B46D48:5B1C1F22
                      
                      {
                        "message": "Not Found",
                        "documentation_url":
                      "https://developer.github.com/v3/repos/releases/#create-a-release"
                      }
                      .
```

--- a/bin/publish.ml
+++ b/bin/publish.ml
@@ -9,8 +9,14 @@ open Dune_release
 
 let gen_doc ~dry_run ~force dir pkg_name =
   let build_doc = Cmd.(v "jbuilder" % "build" % "-p" % pkg_name % "@doc") in
-  let build_dir = Fpath.(dir / "_build" / "default" / "_doc") in
-  let do_doc () = Sos.run ~dry_run ~force build_doc >>| fun () -> build_dir in
+  let doc_dir = Fpath.(dir / "_build" / "default" / "_doc") in
+  let html_dir = Fpath.(doc_dir / "_html") in
+  let do_doc () =
+    Sos.run ~dry_run ~force build_doc >>= fun () ->
+    Sos.dir_exists ~dry_run html_dir >>| function
+    | true  -> html_dir
+    | false -> doc_dir
+  in
   R.join @@ Sos.with_dir ~dry_run dir do_doc ()
 
 let publish_doc ~dry_run pkg =

